servers/rendering/renderer_rd/storage_rd/render_scene_buffers_rd.h

```diff
@@ -0,0 +1,273 @@
+	virtual void set_use_debanding(bool p_use_debanding) override;
+
+	// Named Textures
+
+	bool has_texture(const StringName &p_context, const StringName &p_texture_name) const;
+	RID create_texture(const StringName &p_context, const StringName &p_texture_name, const RD::DataFormat p_data_format, const uint32_t p_usage_bits, const RD::TextureSamples p_texture_samples = RD::TEXTURE_SAMPLES_1, const Size2i p_size = Size2i(0, 0), const uint32_t p_depth = 1, const uint32_t p_layers = 0, const uint32_t p_mipmaps = 1, RD::TextureView p_view = RD::TextureView(), bool p_unique = true);
+	RID create_texture_view(const StringName &p_context, const StringName &p_texture_name, const StringName p_view_name, RD::TextureView p_view = RD::TextureView());
+	RID get_texture(const StringName &p_context, const StringName &p_texture_name) const;
+	const RD::TextureFormat get_texture_format(const StringName &p_context, const StringName &p_texture_name) const;
+	RID get_texture_slice(const StringName &p_context, const StringName &p_texture_name, const uint32_t p_layer, const uint32_t p_mipmap);
+
+	void clear_context(const StringName &p_context);
+
+	// Data buffers
+	bool has_data_buffer(const StringName &p_name);
+	void set_data_buffer(const StringName &p_name, Ref<RenderSceneBufferDataRD> p_buffer);
+	Ref<RenderSceneBufferDataRD> get_data_buffer(const StringName &p_name) const;
+
+	// Getters
+
+	_FORCE_INLINE_ RID get_render_target() const { return render_target; }
+	_FORCE_INLINE_ uint32_t get_view_count() const { return view_count; }
+	_FORCE_INLINE_ Size2i get_internal_size() const { return internal_size; }
+	_FORCE_INLINE_ Size2i get_target_size() const { return target_size; }
+	_FORCE_INLINE_ float get_fsr_sharpness() const { return fsr_sharpness; }
+	_FORCE_INLINE_ RS::ViewportMSAA get_msaa() const { return msaa; }
+	_FORCE_INLINE_ RS::ViewportScreenSpaceAA get_screen_space_aa() const { return screen_space_aa; }
+	_FORCE_INLINE_ bool get_use_taa() const { return use_taa; }
+	_FORCE_INLINE_ bool get_use_debanding() const { return use_debanding; }
+
+	uint64_t get_auto_exposure_version() const { return auto_exposure_version; }
+	void set_auto_exposure_version(const uint64_t p_auto_exposure_version) { auto_exposure_version = p_auto_exposure_version; }
+
+	// For our internal textures we provide some easy access methods.
+
+	_FORCE_INLINE_ RID get_internal_texture() const {
+		return get_texture(SNAME("RenderSceneBuffersRD"), SNAME("internal_texture"));
+	}
+	_FORCE_INLINE_ RID get_internal_texture(const uint32_t p_layer) {
+		return get_texture_slice(SNAME("RenderSceneBuffersRD"), SNAME("internal_texture"), p_layer, 0);
+	}
+
+	_FORCE_INLINE_ RID get_depth_texture() const {
+		return get_texture(SNAME("RenderSceneBuffersRD"), SNAME("depth_texture"));
+	}
+	_FORCE_INLINE_ RID get_depth_texture(const uint32_t p_layer) {
+		return get_texture_slice(SNAME("RenderSceneBuffersRD"), SNAME("depth_texture"), p_layer, 0);
+	}
+
+	// back buffer (color + depth)
```

Comment:
I would probably move these and the ones below away from here since they will only be created when they make sense. If they are not around when needed, and something requests it and its not found, then it should be considered an error and its fine.

Reply:
You mean the back buffer? yeah I can improve on that :)

Reply:
The depth back buffer now creates the buffers if needed in the copy function and where the back buffer is used it's accessed directly.

For the color buffer I still need to redo the blur approach.